README-iso.md

```diff
@@ -0,0 +1,260 @@
+  additional base repo layers.  The `base` layer will always be in the
+  include chain.
+
+XCP-ng official repositories are located at
+https://updates.xcp-ng.org/ and most of them are available through
+standard "repo" layers; e.g. the `testing` repository for `8.2` LTS can
+be used as `8.2:testing`.
+
+Custom repositories can be added with `--define-repo` flag.  They will
+be used by `yum` using the first `CUSTOMREPO.tmpl` template found in
+the layer search path (one is provided in `base`).
+
+## examples
+
+### 8.3 updates and testing
+
+```
+./scripts/mirror-repos.sh 8.3
+
+sudo ./scripts/create-installimg.sh \
+    --srcurl file://$HOME/mirrors/xcpng/8.3 \
+    -o install-8.3-testing.img \
+    8.3:testing
+
+./scripts/create-install-iso.sh \
+    --srcurl file://$HOME/mirrors/xcpng/8.3 \
+    -V "XCP-NG_830_TEST" \
+    8.3:testing install-8.3-testing.img xcp-ng-8.3-testing.iso
+```
+
+### tip of 8.2 (8.2 + updates)
+
+```
+./scripts/mirror-repos.sh 8.2
+
+sudo ./scripts/create-installimg.sh \
+    --srcurl file://$HOME/mirrors/xcpng/8.2 \
+    8.2:updates
+
+./scripts/create-install-iso.sh \
+    --srcurl file://$HOME/mirrors/xcpng/8.2 \
+    -V "XCP-NG_82U_TEST" \
+    8.2:updates install-8.2.updates-x86_64.img xcp-ng-8.2.updates-install.iso
+```
+
+### testing boot modes in qemu
+
+PC BIOS:
+
+```
+qemu-system-x86_64 -serial stdio -m 1G -cdrom xcp-ng-8.3-install.iso
+```
+
+UEFI:
+
+```
+qemu-system-x86_64 -serial stdio -m 1G -cdrom xcp-ng-8.3-install.iso \
+  --bios /usr/share/edk2/ovmf/OVMF_CODE.fd -net none
+```
+
+### testing that scripts run correctly
+
+Minimal tests to generate install ISO for a few important
+configurations are available in `tests/`.  They require one-time
+initialization of the `tests/sharness/` submodule:
+
+```
+git submodule update --init tests/sharness/
+```
+
+They require setting a variable pointing to the repositories to be
+used; it is recommended you use a local mirror for this.
+
+Two common ways of running the tests are:
+
+* just using `make`, which produce human-readable
+  [TAP](http://testanything.org/) output:
+
+  ```
+  make -C tests/ XCPTEST_REPOROOT=file:///data/mirrors/xcpng
+  ```
+
+* through `prove` (in package `perl-Test-Harness` in
+  CentOS/Fedora/RHEL, in `perl` for the rest of the world), which
+  provides both options for e.g. parallel running, and global summary:
+
+  ```
+  XCPTEST_REPOROOT=file:///data/mirrors/xcpng prove tests/
+  ```
+
+The tests for producing `install.img` are tagged as expensive and not
+run by default, to run them you must pass the `-l` flag to the test
+script, which can be achieved respectively by:
+
+```
+make TEST_OPTS="-l"
+
+prove tests/ :: -l
+```
+
+
+## still to be done (a bit outdated but not that much)
+
+* [ ] .treeinfo files have an unsubst'd `{{timestamp}}`
+* [x] how to deal with the fact we can only reproduce 8.x.0 and 8.x.updates
+* [ ] (^^ linked ^^) customizability of available repos
```

Comment:
Done, or there's more to be done about it?

Reply:
right